test/test_common/environment.h

```diff
@@ -21,6 +22,17 @@ class TestEnvironment {
   static void initializeOptions(int argc, char** argv);
 
   /**
+   * @return bool if testing only with IP type addresses only.
+   */
+  static bool isTestIpVersionOnly(const Network::Address::IpVersion& type);
```

Comment:
Maybe `shouldRunTestForIpVersion()`, to avoid confusing with a method that does actually run something. This would imply changing the logic to reflect the name.